src/error.rs

```diff
@@ -284,19 +276,330 @@ impl From<InvalidWindowHandle> for CmdError {
+
+    /// An error occurred while executing JavaScript supplied by the user.
+    JavascriptError,
+
+    /// The target for mouse interaction is not in the browser’s viewport and
+    /// cannot be brought into that viewport.
+    MoveTargetOutOfBounds,
+
+    /// An attempt was made to operate on a modal dialogue when one was not
+    /// open.
+    NoSuchAlert,
+
+    /// No cookie matching the given path name was found amongst the associated
+    /// cookies of the current browsing context’s active document.
+    NoSuchCookie,
+
+    /// An [element] could not be located on the page using the given search
+    /// parameters.
+    ///
+    /// [element]: https://www.w3.org/TR/webdriver1/#dfn-elements
+    NoSuchElement,
+
+    /// A [command] to switch to a frame could not be satisfied because the
+    /// frame could not be found.
+    ///
+    /// [command]: https://www.w3.org/TR/webdriver1/#dfn-commands
+    NoSuchFrame,
+
+    /// An [element]'s [ShadowRoot] was not found attached to the element.
+    ///
+    /// [element]: https://www.w3.org/TR/webdriver2/#dfn-elements
+    /// [ShadowRoot]: https://www.w3.org/TR/webdriver2/#dfn-shadow-roots
+    NoSuchShadowRoot,
+
+    /// A [command] to switch to a window could not be satisfied because the
+    /// window could not be found.
+    ///
+    /// [command]: https://www.w3.org/TR/webdriver1/#dfn-commands
+    NoSuchWindow,
+
+    /// A script did not complete before its timeout expired.
+    ScriptTimeout,
+
+    /// A new session could not be created.
+    SessionNotCreated,
+
+    /// A [command] failed because the referenced [element] is no longer
+    /// attached to the DOM.
+    ///
+    /// [command]: https://www.w3.org/TR/webdriver1/#dfn-commands
+    /// [element]: https://www.w3.org/TR/webdriver1/#dfn-elements
+    StaleElementReference,
+
+    /// An operation did not complete before its timeout expired.
+    Timeout,
+
+    /// A screen capture was made impossible.
+    UnableToCaptureScreen,
+
+    /// Setting the cookie’s value could not be done.
+    UnableToSetCookie,
+
+    /// A modal dialogue was open, blocking this operation.
+    UnexpectedAlertOpen,
+
+    /// The requested command could not be executed because it does not exist.
+    UnknownCommand,
+
+    /// An unknown error occurred in the remote end whilst processing the
+    /// [command].
+    ///
+    /// [command]: https://www.w3.org/TR/webdriver1/#dfn-commands
+    UnknownError,
+
+    /// The requested [command] matched a known endpoint, but did not match a
+    /// method for that endpoint.
+    ///
+    /// [command]: https://www.w3.org/TR/webdriver1/#dfn-commands
+    UnknownMethod,
+
+    /// Unknown WebDriver command.
+    UnknownPath,
+
+    /// Indicates that a command that should have executed properly is not
+    /// currently supported.
+    UnsupportedOperation,
+}
+
+impl ErrorStatus {
+    /// Returns the string serialisation of the error type.
+    pub fn error_code(&self) -> &'static str {
```

Comment:
It's weird to me that this is called error _code_, but returns a string. A code is (at least to me) generally a numerical representation. Do we even need this, or can we just implement `Display` instead?

Reply:
I also think it's probably reasonable to implement `std::error::Error` for this type (which should be trivial since we derive `Debug` and if we implement `Display`).